src/plugins/typedispatcher/typereader.c

```diff
@@ -0,0 +1,422 @@
+#include "typehelper.h"
+#include <kdbease.h>     //KeyGetRelativeName, elektraArrayIncName
+#include <kdbhelper.h>   //elektraCalloc
+#include <kdbproposal.h> //keyRel2, elektraKeyGetMetaKeySet
+#include <stdio.h>
+
+
+// creates skeletons(name+type+scope) for new types and fails if type already exist
+static int readTypeNames (DispatchConfig * config, const Key * scope, const Key * typeKey, Key * parentKey ELEKTRA_UNUSED)
+{
+	if (!typeKey) return ERROR;
+
+	const char * typeName = keyBaseName (typeKey);
+#ifdef DEVBUILD
```

Comment:
i'll drop both of them when i'm done, they are just for internal debugging (+ vim hides/folds them nicely)